packages/x-client/src/config/index.ts

```diff
@@ -174,9 +185,9 @@ export const imxClientConfig = (configOptions: ConfigOptions): ImxModuleConfigur
     throw new Error(`Invalid environment: ${configOptions.environment}`);
   }
 
-  const clientConfig = new ImxConfiguration({
+  const clientConfig = {
     baseConfig: new ImmutableConfiguration(configOptions),
-  });
+  } as ImxModuleConfiguration;
```

Comment:
just curious, why is this "as" matters since you are returning clientConfig anyway?

Reply:
Betting typing for the user. `IMXClient` expects a `ImxModuleConfiguration`, the function returns an object with the required properties, but when inspected it just says it's an object with properties, instead of a `ImxModuleConfiguration`.